app/include/zmk/event_manager.h

```diff
@@ -38,21 +38,24 @@ struct zmk_event_subscription {
         zmk_event_t header;                                                                        \
         struct event_type data;                                                                    \
     };                                                                                             \
-    struct event_type##_event *new_##event_type(struct event_type);                                \
+    struct event_type##_event new_##event_type(struct event_type);                                 \
+    int raise_new_##event_type(struct event_type);                                                 \
```

Comment:
So.... Part of the reason for creating the `RAISE_` macros was because we *can't* raise the event by passing the pointer to it if there isn't actually a local created to create an owner that has a proper scope so it'll be cleared from the stack when the local function exits.

So we can't just pass the result *directly* to the raise function. *But* I'm fine if the `create_` terminology reads better.

Reply:
To clarify this and my previous comment, it looks like the only other thing using the `new_<event>` function is `create_layer_state_changed()`. If `create_layer_state_changed()` were to return a `struct zmk_layer_state_changed` instead of a `struct zmk_layer_state_changed_event`, then you could just pass the result to `raise_new_zmk_layer_state_changed_event()` and have it wrap the data in an event for you, which would simplify the code a bit:

```diff
- struct zmk_layer_state_changed_event ev = create_layer_state_changed(layer, state);
- ZMK_EVENT_RAISE(ev);
+ zmk_raise_layer_state_changed_event(create_layer_state_changed(layer, state));
```

Then, there would be nothing left using `new_<event>` aside from `raise_new_<event>`. Is there any reason to create an event object but *not* immediately raise it? If not, we could remove the `new_<event>` function entirely (and roll its code into the raise function). If so, it might be nice to rename it to avoid the association with C++'s `new` operator, but that's probably not a big deal.

And then for this comment, I meant specifically the "new" in `raise_new_<event>`. There's no "raise *old* event" function to make a distinction between, so I think the name could be simplified to just `raise_<event>`.

Reply:
Indeed. Ok, so renamed the `raise_` function to skip the `new` and removed the `new_` generated functions entirely in favor of the `raise_` being used uniformly. Thanks!